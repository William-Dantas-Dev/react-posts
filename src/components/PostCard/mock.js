export const postCardPropsMock = {
  title: 'test title',
  body: 'test body',
  id: 1,
  cover: 'img/img.png'
}
