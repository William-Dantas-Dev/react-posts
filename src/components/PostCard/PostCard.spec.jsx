import { render, screen } from '@testing-library/react';
import { PostCard } from '.';
import { postCardPropsMock } from './mock';

const props = postCardPropsMock;

describe('<PostCard />', () => {
  it('should render PostCard correctly', () => {
    render(<PostCard {...props} />);

    expect(screen.getByAltText(/test title/i))
      .toHaveAttribute('src', 'img/img.png');
    expect(screen.getByRole('heading', { name: 'test title 1' })).toBeInTheDocument();
    expect(screen.getByText('test body')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<PostCard {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});