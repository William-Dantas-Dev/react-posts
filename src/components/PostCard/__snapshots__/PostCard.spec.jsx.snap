// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="test title"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h2>
      test title
       
      1
       
    </h2>
    <p>
      test body
    </p>
  </div>
</div>
`;
